index.js

```diff
@@ -42,9 +45,20 @@ const portCheckSequence = function * (ports) {
 
 module.exports = async options => {
 	let ports;
+	let exclude = new Set();
 
 	if (options) {
-		ports = typeof options.port === 'number' ? [options.port] : options.port;
+		if (options.port) {
+			ports = typeof options.port === 'number' ? [options.port] : options.port;
+		}
+
+		if (options.exclude) {
+			if (typeof options.exclude[Symbol.iterator] !== 'function') {
+				throw new TypeError('Exclude option must be set as a valid iterator ( for example using "makeRange" )');
+			}
+
```

Comment:
It should also validate that all the `exclude` elements are numbers.

Reply:
thanks for the feedback, I have added a loop to check through each value, let me know if there is a more elegant way.